src/eth1spec/rlp.py

```diff
@@ -351,3 +354,29 @@ def decode_item_length(encoded_data: Bytes) -> int:
         )
 
     return 1 + length_length + decoded_data_length
+
+
+#
+# Encoding and decoding dataclasses
+#
+
+
+def encode_dataclasses(raw_dataclass_obj: Any) -> Bytes:
+    """
+    Encoding dataclasses. One should never invoke these methods directly.
+    Rather use the encode function defined above
+    """
+    return encode(astuple(raw_dataclass_obj))
+
+
+def decode_to_dataclasses(encoded_data: Bytes, klass: Any) -> Any:
```

Comment:
Similarly, I think decoding should be in an `account_from_tuple(decoded: Tuple[RLP])`, but I could certainly be convinced otherwise.

Reply:
I have currently not implemented the decoding to a dataclass like Account, etc. as we aren't using the decoding anywhere now. Regardless I would have implemented this, except for the fact that on decoding the `Account` rlp encoding, we get the `code_hash` and `storage_root` instead of `code` and `storage` which is what is stored in the dataclass, which is causing an inconsistency wrt the other dataclasses. Should we consider restructuring the `Account` dataclass?

\cc @lightclient

Reply:
I think at some point we may want to create a dataclass like `RawAccount` or something that has all immediate. For now, I think prefer to just remove this decode function.

Reply:
Makes sense.